src/accel/InitializeActions.cc

```diff
@@ -0,0 +1,72 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2022 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file accel/InitializeActions.cc
+//---------------------------------------------------------------------------//
+#include "InitializeActions.hh"
+
+#include <G4RunManager.hh>
+
+#include "corecel/Assert.hh"
+#include "corecel/sys/Device.hh"
+
+#include "EventAction.hh"
+#include "RunAction.hh"
+#include "SetupOptions.hh"
+#include "SharedParams.hh"
+#include "TrackingAction.hh"
+
+namespace celeritas
+{
+namespace
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct actions on each worker thread for Celeritas offloading.
+ */
+template<class Manager>
```

Comment:
Note that this extra templating was to try to support templating on `G4EventManager` which CMSSW uses. However, it won't work as-is because there's no `RunAction` there...

Reply:
Remind me why we can't just use the `RunManager` to set the user actions (since it calls `EventManager::SetUserAction()` under the hood)?

Reply:
Not sure I understand the question.

Originally we were overriding the `G4VUserActionInitialization` until it turned out there can only be one. Then we said OK let's call the `RunManager` since that's all the action initialization does. But CMS directly calls the `EventManager` which has a bunch of `SetUserAction` functions, and then I found out it does *not* have a `UserRunAction`. 

Right now it basically *is* calling the run manager as you say, but we haven't figured out how to make it work with CMS's direct calls to the event manager.

Are you simply asking me to un-template this since it's only used in the one instantiation? 😅

Reply:
Discussed offline and I think I understand now.

Reply:
(In summary, I think we can't directly interface with the `G4RunManager` because CMSSW doesn't seem to use one: it has a different "run manager" class that does most of the same things but has a different interface.)